File: backend/src/tokens.js
const crypto = require("crypto");

const createToken = () => {
  const token = `${crypto.randomBytes(64).toString("base64")}}`;
  return token;
}

module.exports.createToken = createToken;

const createPassword = () => {
  const password = `${crypto.randomBytes(16).toString("base64")}}`;
  return password;
}

module.exports.createPassword = createPassword;

/** @param {string} date */
const daysFrom = (date) => {
  const formatDate = new Date(date);
  const currentDate = new Date();
  const differenceInTime = currentDate.getTime() - formatDate.getTime();
  const differenceInDays = differenceInTime / (1000 * 3600 * 24);
  return Math.abs(Math.round(differenceInDays));
}

/** @param {string} date */
const dateIsValid = (date) => {
  const days = daysFrom(date);
  return days <= 10;
}

module.exports.dateIsValid = dateIsValid;

File: backend/src/tokens.py
import secrets
import base64
from db import get_db
from datetime import datetime

def generate_token():
  random_bytes = secrets.token_bytes(64)
  token = base64.b64encode(random_bytes).decode("utf-8")

  db = get_db()

  cursor = db.cursor()
  cursor.execute("INSERT INTO tokens (token) VALUES (?)", (token,))
  db.commit()

  cursor.close()

  return token

def days_from(date: str):
  return (datetime.now() - datetime.fromisoformat(date)).days

def date_is_valid(date: str):
  return days_from(date) >= 10

def generate_password():
  random_bytes = secrets.token_bytes(16)
  password = base64.b64encode(random_bytes).decode("utf-8")

  db = get_db()

  cursor = db.cursor()
  cursor.execute("INSERT INTO tempPasswords (password) VALUES (?)", (password,))
  db.commit()

  cursor.close()

  return password